shared/data/src/main/java/com/ivy/data/di/RemoteDataSourceModule.kt

```diff
@@ -0,0 +1,20 @@
+package com.ivy.data.di
+
+import com.ivy.data.remote.RemoteExchangeRatesDataSource
+import com.ivy.data.remote.impl.RemoteExchangeRatesDataSourceImpl
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import io.ktor.client.HttpClient
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object RemoteDataSourceModule {
+    @Provides
+    @Singleton
+    fun provideRemoteExchangeRatesDataSource(ktorClient: dagger.Lazy<HttpClient>): RemoteExchangeRatesDataSource {
+        return RemoteExchangeRatesDataSourceImpl(ktorClient)
```

Comment:
Why this provides? Use `@Binds` + `@Inject constructor`

Reply:
See how the other dataources or repositories are implemented:
- none instances are created manually by calling the constructor
- Hilt `@Binds`
- `@Inject constructor`
- this way if I add more params to the impl.class constructor it'll work out of the box and I won't have to modify some provide methods

Reply:
Oh okay I get it now